enhancements/olm-toggle-copied-csvs.md

```diff
@@ -0,0 +1,163 @@
+
+### Goals
+
+- Allieviate the resource utilization incurred by Copied CSVs
+- Don't block upgrades for customers of OpenShift
+- Provide an "escape hatch" that can be backported in Z-stream releases to supported minor versions of OpenShift
+
+### Non-Goals
+
+- Provide a replacement for Copied CSVs
+
+## Proposal
+
+For each non-Copied CSV, a single [Operator](https://github.com/operator-framework/enhancements/blob/master/enhancements/simplify-olm-apis.md#proposal) resource will be generated. This cluster-scoped resource will take the place of any related Copied CSVs.
+
+### User Stories
+
+#### Toggle Copied CSVs Off
+
+As a Cluster Admin, I want a way to disable Copied CSVs so that I can reduce resource consumption ahead of the feature's deprecation and eventual removal.
+
+#### Toggle Copied CSVs On
+
+As a Cluster Admin, I want a way to enable Copied CSVs if I've reconfigured the cluster in ways that mitigate the problem; e.g. uninstalled operators from `AllNamespace` `OperatorGroups`.
+
+### Implementation Details/Notes/Constraints
+
+#### Operator Cardinality
+
+Today, Operator resources are generated on a per-Subscription basis. Since CSVs -- and thus Copied CSVs -- can exist without matching Subscriptions, OLM's controllers will be extended to generate additional Operator resources for these "free CSVs" such that:
+
+- there exists at least one`Operator` resource for every CSV
+- there exists at least one `Operator` resource for every `Subscription`
+
+That is, all CSVs installed by a given `Subscription` share an `Operator` resource __and__ any CSVs that weren't installed by a `Subscription` are related to their own `Operator` resource.
+
+#### Orphaned CSVs
+
+Orphaned CSVs are CSVs that were installed via a `Subscription` that no longer exists. Since the lifetimes -- i.e. the period of existence on a cluster --  of `Operator` resources aren't directly tied to the lifetimes of the resources that triggered their generation, the `Operator` resource(s) associated with Orphaned CSVs will continue to exist _after_ a `Subscription` is deleted. If that `Operator` resource is deleted explicitly, OLM will generate another as per the rules [outlined above](#operator-cardinality).
+
+#### Copied RBAC 
+
+In addition to operator discoverability, Copied CSVs also serve as convienient vehicles for the garbage collection of Copied RBAC. These are resources that are copied along with the CSV to provide the operator access to a namespace. To support the loss of Copied CSVs, the RBAC copying logic in OLM will be extended to write additional owner references from copied resources to the related Operator resource. 
+
+#### The Toggle
+
+A feature toggle will be added to control OLM's use of Copied CSVs. When enabled (toggled on), OLM will delete all existing Copied CSVs and will not generate any more. When disabled (toggled off), OLM will generate Copied CSVs (much like it does today).
```

Comment:
We should tread lightly here. Copied CSVs are powering one of the major value-add of OLM which is operator discoverability by under-privileged tenants on the cluster (those without `list` on CRD). Removing copied CSVs severely degrades their experience. Especially the UIs that are based on Copied CSVs will become non-functional for tenants.

This should be reduced to disabling Copied CSVs of operators that are installed cluster-wide, i.e. who's associated `OperatorGroup` operates in `AllNamespace` mode.

Reply:
> This should be reduced to disabling Copied CSVs of operators that are installed cluster-wide, i.e. who's associated OperatorGroup operates in AllNamespace mode.

Fair enough.

Reply:
I added a mitigation to a later section that proposes the toggle only affect `AllNamespace` mode installations.